### PR TITLE
Closes #534 | Add/Update Dataloader idner_news_2k

### LINKED ISSUE #534: Create dataset loader for idner-news-2k
Dataloader name: `idner_news_2k/idner_news_2k.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?idner_news_2k

|     Dataset| idner_news_2k |
|-------------|---|
| Description | A dataset of Indonesian News for Named-Entity Recognition task. This dataset was previously provided by Syaifudin & Nurwidyantoro (2016) (https://github.com/yusufsyaifudin/Indonesia-ner). We manually re-annotated the dataset with a more standardized NER tags. We split this dataset into three files, namely train.txt, dev.txt, and test.txt. Each file consists of three columns which are Tokens, PoS Tag, and NER Tag respectively. The format is following CoNLL dataset which split each token into one line and each sentence is separated by one empty line. For the NER tag, we use the IOB format as illustrated in the example below. In terms of PoS tag, we tagged the data using UDPipe (http://ufal.mff.cuni.cz/udpipe), a pipeline for tokenization, tagging, lemmatization and dependency parsing whose model is trained on UD Treebanks. |
| Subsets | - |
| Languages | ind |
| Tasks | Named Entity Recognition |
| License | MIT (mit) |
| Homepage | https://github.com/khairunnisaor/idner-news-2k |
| HF URL | - |
| Paper URL | https://aclanthology.org/2020.aacl-srw.10/ |

### PR DESCRIPTION
**Pull request by R-Damanhuri**
Closes #534 

### Checkbox
- [X] Confirm that this PR is linked to the dataset issue.
- [X] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [X] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [X] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [X] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [X] Confirm dataloader script works with `datasets.load_dataset` function.
- [X] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/idner_news_2k/idner_news_2k.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+"""
+A dataset of Indonesian News for Named-Entity Recognition task.
+  This dataset re-annotated the dataset previously provided by Syaifudin & Nurwidyantoro (2016)
+  (https://github.com/yusufsyaifudin/Indonesia-ner) with a more standardized NER tags.
+  There are three subsets, namely train.txt, dev.txt, and test.txt.
+  Each file consists of three columns which are Tokens, PoS Tag, and NER Tag respectively.
+  The format is following CoNLL dataset. The NER tag use the IOB format.
+  The PoS tag using UDPipe (http://ufal.mff.cuni.cz/udpipe),
+  a pipeline for tokenization, tagging, lemmatization and dependency parsing
+  whose model is trained on UD Treebanks.
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/idner_news_2k/idner_news_2k.py:25]: I suggest including this in the `_DESCRIPTION`

File: seacrowd/sea_datasets/idner_news_2k/idner_news_2k.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        urls = _URLS[_DATASETNAME]
+        train_path = dl_manager.download_and_extract(urls["train"])
+        dev_path = dl_manager.download_and_extract(urls["dev"])
+        test_path = dl_manager.download_and_extract(urls["test"])
+
+        # Not all datasets have predefined canonical train/val/test splits.
+        # If your dataset has no predefined splits, use datasets.Split.TRAIN for all of the data.
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/idner_news_2k/idner_news_2k.py:158]:
```suggestion
```

File: seacrowd/sea_datasets/idner_news_2k/idner_news_2k.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+                if pd.isnull(row[0]):
+                    if tokens:
+                        yield idx, {"id": idx, "tokens": tokens, "labels": ner_tags}
+                        tokens, ner_tags = [], []
+                else:
+                    tokens.append(row[0])
+                    ner_tags.append(row[2])
+
+# This template is based on the following template from the datasets package:
+# https://github.com/huggingface/datasets/blob/master/templates/new_dataset_script.py
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/idner_news_2k/idner_news_2k.py:214]:
```suggestion
```

File: seacrowd/sea_datasets/idner_news_2k/idner_news_2k.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+                {
+                    "tokens": datasets.Sequence(datasets.Value("string")),
+                    "pos_tags": datasets.Sequence(datasets.ClassLabel(names=POS_TAGS)),
+                    "ner_tags": datasets.Sequence(datasets.ClassLabel(names=NAMED_ENTITIES)),
+                }
+            )
+
+        elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            features = schemas.seq_label.features(NAMED_ENTITIES)
+
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/idner_news_2k/idner_news_2k.py:140]: maybe an `else` statement that raises a `ValueError`?

**Review by ljvmiranda921 — COMMENTED** [review]
Have a few comments but the dataloader works!

**Review by ljvmiranda921 — APPROVED** [review]
LGTM

**Comment by akhdanfadh** [comment]
LGTM! Merging this now. Thank you for the dataloader @R-Damanhuri 😄

**Pull request merged**